### PR TITLE
Include hip_runtime.h before <complex> for hip-clang

### PR DESCRIPTION
**Pull request by yxsamliu**

### CODE REVIEW

**Comment by ROCm-Apps-Test** [comment]
Can one of the admins verify this patch?

**Comment by ROCm-Apps-Test** [comment]
Can one of the admins verify this patch?

File: tensorflow/core/framework/numeric_types.h [hunk 1]
```diff
@@ -16,6 +16,11 @@ limitations under the License.
 #ifndef TENSORFLOW_FRAMEWORK_NUMERIC_TYPES_H_
 #define TENSORFLOW_FRAMEWORK_NUMERIC_TYPES_H_
 
+#ifdef __HIPCC__
```
[the remark this answers is not in the document]
Comment (whchung) [reply tensorflow/core/framework/numeric_types.h]: I see. Could you explain the rationale why we need to include hip_runtime.h before <complex> for supporting std::complex with hip-clang? Having to depend on a certain order to include header files would certainly be frowned upon during upstreaming.

**Comment by yxsamliu** [comment]
@whchung HIP by design requests user to explicitly include hip/hip_runtime.h. I think it is better not to automatically include hip/hip_runtime.h by default. Instead, this feature should be enabled either with an environment variable or with a command line option. Also it makes more sense to implement this in hipcc instead of clang since hip/hip_runtime.h belongs to HIP runtime. This is different from cuda-clang, where the header files are implemented in clang.

That said, I would suggest to introduce environment variable HIP_AUTO_INCLUDE_HEADER to control hipcc whether to include hip/hip_runtime.h automatically. When HIP_AUTO_INCLUDE_HEADER is set to a non-zero value, hipcc will include hip/hip_runtime.h at the beginning of each HIP source code.

**Comment by whchung** [comment]
@yxsamliu Thanks. I think it's a sound approach to let `hipcc` include `hip_runtime.h`. Closing this PR hereby.

**Pull request closed without merging**